### PR TITLE
gossip: transaction requester thread

### LINKED ISSUE #590: Implement a transaction requester worker
Transaction requests are currently bound to incoming new packets from the gossip protocol.
The ratio amount of tx request / amount of new packets is not enough to keep up with the amount of requests.
The size of the request queue is always at its maximum (10000 being the default value).

### PR DESCRIPTION
**Pull request by thibault-martinez**
Fixes #590 

# Test Plan:
CI

### CODE REVIEW

File: gossip/components/transaction_requester.c [hunk 1]
```diff
@@ -8,189 +8,298 @@
...
+                      .num_bytes = FLEX_TRIT_SIZE_243,
+                      .dynamic = 0};
+  DECLARE_PACK_SINGLE_TX(tx, txp, pack);
+
+  if (transaction_requester == NULL) {
+    return NULL;
+  }
+
+  while (transaction_requester->running) {
+    if (requester_size(transaction_requester) > REQUESTER_THRESHOLD) {
```
Comment (th0br0) [inline gossip/components/transaction_requester.c]: let's use a label here and invert the condition?

**Review by th0br0 — CHANGES_REQUESTED** [review]

File: gossip/components/transaction_requester.c [hunk 2]
```diff
@@ -8,189 +8,298 @@
...
+        if (ret == RC_OK && pack.num_loaded != 0) {
+          transaction_serialize_on_flex_trits(txp, transaction);
+        } else {
+          memset(transaction, FLEX_TRIT_NULL_VALUE, FLEX_TRIT_SIZE_8019);
+        }
+      }
+      if (transaction_requester->node->neighbors) {
+        iter = transaction_requester->node->neighbors->front;
+        while (iter) {
+          if (neighbor_send(transaction_requester->node, &iter->data,
```
Comment (th0br0) [inline gossip/components/transaction_requester.c]: is this thread safe?

Comment (thibault-martinez) [reply gossip/components/transaction_requester.c]: Not 100% thread-safe, I planned on refactoring the whole neighbour layer when doing `getNeighbors`, `addNeighbors` and `removeNeighbors` API calls (maybe next)

Comment (thibault-martinez) [reply gossip/components/transaction_requester.c]: Fixed in #599

**Review by th0br0 — APPROVED** [review]

**Review by th0br0 — APPROVED** [review]

**Pull request merged**